measure.rb

```diff
@@ -663,6 +671,48 @@ def self.check_for_errors(runner, model)
     return true
   end
 
+  def self.modify_cond_basement_surface_properties(runner, model)
+    # modify conditioned basement surface properties here
+    # zero out solar absorptance in conditioned basement
+    @cond_bsmnt_surfaces.each do |cond_bsmnt_surface|
+      # zero out solar absorptance for some surfaces
+      mat_share = false
+      const = cond_bsmnt_surface.construction.get
+      exterior_material = const.to_LayeredConstruction.get.layers[0].to_StandardOpaqueMaterial.get
+      mat_share = if_share_mat(model, cond_bsmnt_surface, exterior_material)
+      if mat_share
+        # create new construction for these surfaces
+        new_const = const.clone.to_Construction.get
+        cond_bsmnt_surface.setConstruction(new_const)
+        exterior_material = exterior_material.clone.to_StandardOpaqueMaterial.get
+        new_const.to_LayeredConstruction.get.setLayer(0, exterior_material)
+      end
+      exterior_material.setSolarAbsorptance(0.0)
+      exterior_material.setVisibleAbsorptance(0.0)
+    end
+    return true
+  end
+
+  def self.if_share_mat(model, surface1, mat1)
```

Comment:
I think you can simply use `mat1.directUseCount > 1` to see if the material is being used by multiple objects, which would obviate the need for this method. But I haven't tested it...

Reply:
Thanks for pointing to this function! It's good to use with an extra same check for construction to avoid this edge case: material is attached once to one construction, but this construction is attached to multiple objects. I'm testing it!